Boolean.CSharp.Test/ExtensionTests.cs

```diff
@@ -11,20 +11,69 @@ namespace Boolean.CSharp.Test
     [TestFixture]
     public class ExtensionTests
     {
-        private Extension _extension;
-        public ExtensionTests()
+        private UserAccount _userAccount;
+        private Bank _bank;
+
+        private Management _management;
+
+        [SetUp]
+        public void Setup()
+        {
+            _management = new Management();
+            _bank = new Bank();
+            _userAccount = UserAccount.CreateAccount("John", "Current", "London");
+        }
+
+
+        [Test]
+        public void Deposit_WhenCalled_DepositsAmount()
+        {
+            _bank.Deposit("John", "Current", "London", 200);
+            Assert.AreEqual(200, _bank.GetBalance("John", "Current", "London"));
+        }
+
+        [Test]
+        public void Withdraw_WhenCalled_WithdrawsAmount()
         {
-            _extension = new Extension();
+            _bank.Deposit("John", "Current", "London", 200);
+            _bank.Withdraw("John", "Current", "London", 100);
+            Assert.AreEqual(100, _bank.GetBalance("John", "Current", "London"));
         }
+
         [Test]
-        private void TestQuestion1()
+        public void WithdrawsAmount()
         {
+            _bank.Deposit("John", "Current", "London", 200);
+            _bank.Withdraw("John", "Current", "London", 300);
+            Assert.AreEqual(-100, _bank.GetBalance("John", "Current", "London"));
+        }
 
+        [Test]
+        public void DoesNotWithdrawAmount()
+        {
+            _bank.Deposit("John", "Current", "London", 200);
+            _bank.Withdraw("John", "Current", "London", 1300);
+            Assert.AreEqual(200, _bank.GetBalance("John", "Current", "London"));
         }
+
         [Test]
-        private void TestQuestion2()
+        public void OverDraftLimit_WhenCalled_ReturnsTrue()
```

Comment:
this test is weak: it doesn't test whetheer the bank account obeys the overdraft if it gets created